GetLatestMigrationsStatus/handler.ts

```diff
@@ -65,21 +86,51 @@ export const createSqlStatus = (dbConfig: IDecodableConfigPostgreSQL) => (
     .as("t")
     .toQuery() as NonEmptyString;
 
-export const createHandler = (): Handler => (): ReturnType<Handler> =>
+export const getLatestMigrationByOrganizationFiscalCode = (
+  config: IConfig,
+  connect: Pool
+) => (
+  organizationFiscalCode: OrganizationFiscalCode
+): TE.TaskEither<IDbError, LatestMigrationResultSet> =>
+  pipe(
+    createSqlStatus(config)(organizationFiscalCode, SubscriptionStatus.INITAL),
+    sql => queryDataTable(connect, sql),
+    TE.mapLeft(flow(toPostgreSQLErrorMessage, toPostgreSQLError))
+  );
+
+export const processResponseFromLatestMigrationResultSet = (
+  resultSet: LatestMigrationResultSet
+): TE.TaskEither<
+  IResponseErrorInternal | IResponseErrorNotFound,
+  IResponseSuccessJson<MigrationsStatus>
+> =>
+  pipe(
+    resultSet,
+    LatestMigrationResultSet.decode,
+    TE.fromEither,
+    TE.mapLeft(() => ResponseErrorInternal("Error on decode")),
+    TE.map(data => ResponseSuccessJson(data.rows))
+  );
+
+export const createHandler = (config: IConfig, pool: Pool): Handler => async (
+  _context,
+  organizationFiscalCode
+): ReturnType<Handler> =>
   pipe(
-    TE.throwError<
-      string,
-      IResponseSuccessJson<{ readonly data: MigrationsStatus }>
-    >("To be Implementend"),
-    TE.mapLeft(ResponseErrorInternal),
+    getLatestMigrationByOrganizationFiscalCode(
+      config,
+      pool
+    )(organizationFiscalCode),
+    TE.mapLeft(flow(E.toError, e => ResponseErrorInternal(e.message))),
```

Comment:
This `mapLeft` takes a `IDbError` object. Do you really need to convert it into an `Error`  first and then into a `ResponseErrorInternal`? Being the incoming value strongly typed, you can skip the `Error` and just compose the error message by yourself.

More: are you sure you aren't losing any information by using the generic `toError` helper? Does the generated `Error` contains in its message all info from `IDbError` that may be useful for troubleshooting? What does the `Error` contains if you pass a `IDbError` to `toError`? (my bet: `[Object object]`).

Reply:
Maybe this can be an improvement 32c96c7?

Reply:
You don't need to pipe the value, just use it
```ts
    TE.mapLeft(error =>
      ResponseErrorInternal(
        `getLatestMigrationByOrganizationFiscalCode|${error.kind}: ${error.message}`
      )
    ),
```